Made overlay_mask work

reallybigname opened the pull request:
Overlay mask working properly with static mask or video mask.
- made new masks.py file that we can use to consolidate other mask functions from main render code
- Overlay mask NOW WORKS WITH CADENCE.  I made the function switchable between PIL rgb and np bgr and inject the function during cadence and for normal saves
- I didn't use old code. I just made this work myself.

Fixed one-off issue with hybrid video where the first frame was duplicated. (it became obvious when doing overlay video masking)

re-commented a few lines, removed some whitespace

reallybigname committed:
Made overlay_mask work

Overlay mask working properly with static mask or video mask.
- made new masks.py file that we can use to consolidate other mask functions from main render code
- Overlay mask NOW WORKS WITH CADENCE.  I made the function switchable between PIL rgb and np bgr and inject the function during cadence and for normal saves
- I didn't use old code. I just made this work myself.

Fixed one-off issue with hybrid video where the first frame was duplicated. (it became obvious when doing overlay video masking)

re-commented a few lines, removed some whitespace

reallybigname committed:
Merge branch 'automatic1111-webui' of https://github.com/reallybigname/deforum-for-automatic1111-webui into automatic1111-webui

hithereai commented:
Default 2d settings, just clicked on use_mask in init tab, and got this (note that when un-selecting overlay mask it works):
```
Traceback (most recent call last):
  File "D:\D-SD\AUTO04\stable-diffusion-webui\modules\call_queue.py", line 56, in f
    res = list(func(*args, **kwargs))
  File "D:\D-SD\AUTO04\stable-diffusion-webui\modules\call_queue.py", line 37, in f
    res = func(*args, **kwargs)
  File "D:\D-SD\AUTO04\stable-diffusion-webui\extensions\deforum-for-automatic1111-webui\scripts\deforum.py", line 100, in run_deforum
    render_animation(args, anim_args, video_args, parseq_args, loop_args, controlnet_args, root.animation_prompts, root)
  File "D:\D-SD\AUTO04\stable-diffusion-webui/extensions/deforum-for-automatic1111-webui/scripts\deforum_helpers\render.py", line 560, in render_animation
    image = do_overlay_mask(args, anim_args, image, frame_idx)
  File "D:\D-SD\AUTO04\stable-diffusion-webui/extensions/deforum-for-automatic1111-webui/scripts\deforum_helpers\masks.py", line 17, in do_overlay_mask
    current_mask = Image.open(args.mask_file)
  File "D:\D-SD\AUTO04\stable-diffusion-webui\venv\lib\site-packages\PIL\Image.py", line 3227, in open
    fp = builtins.open(filename, "rb")
OSError: [Errno 22] Invalid argument: 'https://deforum.github.io/a1/M1.jpg'

Traceback (most recent call last):
  File "D:\D-SD\AUTO04\stable-diffusion-webui\venv\lib\site-packages\gradio\blocks.py", line 987, in postprocess_data
    if predictions[i] is components._Keywords.FINISHED_ITERATING:
IndexError: tuple index out of range

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "D:\D-SD\AUTO04\stable-diffusion-webui\venv\lib\site-packages\gradio\routes.py", line 394, in run_predict
    output = await app.get_blocks().process_api(
  File "D:\D-SD\AUTO04\stable-diffusion-webui\venv\lib\site-packages\gradio\blocks.py", line 1078, in process_api
    data = self.postprocess_data(fn_index, result["prediction"], state)
  File "D:\D-SD\AUTO04\stable-diffusion-webui\venv\lib\site-packages\gradio\blocks.py", line 991, in postprocess_data
    raise ValueError(
ValueError: Number of output components does not match number of values returned from from function f
```

hithereai committed:
Merge remote-tracking branch 'upstream/automatic1111-webui' into pr/546

reallybigname committed:
made overlay masks with online images

now loads init_image and mask_file using load_image function

also fixed issue where you needed an init_image
  - passes image through if no init image

reallybigname committed:
Merge branch 'automatic1111-webui' of https://github.com/reallybigname/deforum-for-automatic1111-webui into automatic1111-webui
